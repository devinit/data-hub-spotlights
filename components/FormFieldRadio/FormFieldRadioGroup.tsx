import React, { FunctionComponent } from 'react';

interface ComponentProps {
  label: string;
  value: string | string[] | number;
  name: string;
  checked?: boolean;
  onClick?: (value?: string | string[] | number) => void;
}

const FormFieldRadioGroup: FunctionComponent<ComponentProps> = ({ label, name, value, checked, ...props }) => {
  const onClick = (): void => {
    if (props.onClick) {
      props.onClick(value);
    }
  };

  return (
    <div className="form-radio-group cp-form-radio-group-share" onClick={onClick}>
      <input
        className="form-radio"
        type="radio"
        name={name}
        defaultValue={value}
        checked={checked}
        onChange={onClick}
      />
      <label className="form-label" htmlFor={name}>
        {label}
      </label>
    </div>
  );
};

export { FormFieldRadioGroup };
