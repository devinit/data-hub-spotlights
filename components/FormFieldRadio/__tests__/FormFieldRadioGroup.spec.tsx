/**
 * @jest-environment jsdom
 */
import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { FormFieldRadioGroup } from '../FormFieldRadioGroup';

describe('FormFieldRadioGroup', () => {
  test('renders correctly', () => {
    const renderer = TestRenderer.create(<FormFieldRadioGroup label="this" name="that" value="this-that" />).toJSON();

    expect(renderer).toMatchSnapshot();
  });
});
