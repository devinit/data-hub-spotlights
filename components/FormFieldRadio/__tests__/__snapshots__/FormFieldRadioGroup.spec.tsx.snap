// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormFieldRadioGroup renders correctly 1`] = `
<div
  className="form-radio-group cp-form-radio-group-share"
  onClick={[Function]}
>
  <input
    className="form-radio"
    defaultValue="this-that"
    name="that"
    type="radio"
  />
  <label
    className="form-label"
    htmlFor="that"
  >
    this
  </label>
</div>
`;
